### ISSUE TITLE
[Feature] Set default legend icon for different series types

### ISSUE THREAD

**Issue by andyfehrenbach** [apache/echarts#20339]
### What problem does this feature solve?

I have not been able to figure out a way to do this. I think you should be able to declare a default icon type in the legend for each different series, for example:
bar series: star icon
line series: triangle icon
I know that setting a different symbol on a line series will change the icon in the legend, but there is no way to do this for bar charts

### What does the proposed API look like?

bar: {
   icon: 'star',
},
line: {
   icon: 'triangle',
}

**Comment by Ovilia** [comment]
The icon in the legend represents that of that series. So it doesn't always make sense to use the same icon for the same series.

**Comment by andyfehrenbach** [comment]
I agree that being able to assign different icons to different series of the same chart type, either via series symbols or the legend's data array, is a desired behavior. But I think there is also value in being able to choose the default or fallback icon for a given series type. The specific problem I'm trying to solve in my app is that the rounded rectangle legend elements don't look good with our design language. I would like them to be normal rectangles or squares. but if I set that in the theme file using the legend's icon property, it also changes the icon for all line series, even though I've defined the 'emptyCircle' symbol for line series.

**Comment by helgasoft** [comment]
> I set the icon property to 'rectangle' in the theme file

please provide code example for your _theme_ problem.

> setting a different symbol on a line series will change the icon in the legend, but there is no way to do this for bar charts

setting legend icons is very flexible, even for bars - [Demo](https://echarts.apache.org/examples/en/editor.html?c=dataset-simple1&code=A4ZwLghgTgBAvAKBjA5AWQIwFYB0AWAJgIBoMAOHDAgTgBsBaXAgNj3oJwHZmyGnmsxDt14YcABnHNGOMswDMAGTGtiFDBjy1cZMp3rjZWLDFocFBGfPlbz14rmtadZQYddYGeHASxli3tgEyriEJOSUNABeKADcCODQsHCowBBgABYAXAD0OSgwANQwoJBQ8QAm6RAgAKZg8DAA2kgwAN6tyBUAlgC2tQB2IN0A9kNZzSjAUCMVAK4AxmAoxKgE4tgraxvMWyjrGJwr-xtkKAC6xJ0wICNzUAu1Ey3Ir5No6QsZEDCK6WC1LZ4eQ4eSrVyyVbUEGcIQES7XZBNdDdWgAaxgABVahAtmQQRhVpwQXhVsZKMRrAi3kiUABhDK1Wp1GB0kYLEa48HMfCrAQ-cEcQQaak05EAdQgtAGcwaACEZgB3AbdQFEjikmBYEHUVbyaiUc7XI3IAC-CHO8VotQA5hVGi92jABhB-hMTptVt0OQN3T1XWMKgVTVdkG1na6ntsMLsvT73WAoN0IAMbdbg6GnS63dGjnGxu6oLUlhnWuHs1GPWd876bmVYObLQgRsAwKMBo0OsgwCMRrQ28AJm0Q60qpA6mBMwAPACCU-6ICHMDAAE9gJWFulbSMoCvS8gV3OF0OR8hrTbBhUl2OIBNz_aYKebrUk8znuHV-v3QAjaAZp2fpWv5QP-H5rpWtDdAMgKPqsYFfqgkHQVsIArr0359lkiRQFAqyoehfYAMrdFETxEI-RqmrEQA)
![image](https://github.com/user-attachments/assets/590bf1d1-3a6e-4478-bb61-a06b5a618c64)
